osu.Framework/Graphics/UserInterface/TabControl.cs

```diff
@@ -240,7 +242,15 @@ protected virtual void RemoveTabItem(TabItem<T> tab, bool removeFromDropdown = t
                 throw new InvalidOperationException($"Cannot remove non-removable tab {tab}. Ensure {nameof(TabItem.IsRemovable)} is set appropriately.");
 
             if (tab == SelectedTab)
-                SelectedTab = null;
+            {
+                if (Items.Count == 1)
+                    SelectedTab = null;
+                else
+                {
+                    tabMap.TryGetValue(Items.LastOrDefault(), out TabItem<T> lastTab);
```

Comment:
I'm internally debating whether this shouldn't be
```suggestion
                    tabMap.TryGetValue(Items.Last(), out TabItem<T> lastTab);
```
Reasoning is:

* Right now rider complains (rightly) that `TryGetValue` could throw due to trying to pass a null key.
* `RemoveTabItem` should never get called if `Items.Count == 0`.

This should never fail, but I think dropping the `OrDefault` would make for a saner (and easier to debug) failure.